pkg_resources/__init__.py

```diff
@@ -82,8 +104,8 @@
 
 # declare some globals that will be defined later to
 # satisfy the linters.
-require = None
-working_set = None
+require = cast("Callable[..., Any]", None)
+working_set = cast("WorkingSet", None)
 add_activation_listener = None
 cleanup_resources = None
 resource_stream = None
```

Comment:
I feel like the way it's been done in
https://github.com/pypa/setuptools/pull/4242/files#diff-e3d3d454fa3a072c9f46f8affa27513892fbc2d245e87f57a5927a7be851de05R3332-R3349
is better than casting all these with forward references and redefining callable types.